### PR TITLE
Bring `pipelineOperator` flag in line with minimal

### PR DESCRIPTION
**Pull request by mAAdhaTTah**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | N/A
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  | Yes
| Minor: New Feature?      | No
| Tests Added + Pass?      | Yes
| Documentation PR         | No
| Any Dependency Changes?  | No
| License                  | MIT

The minimal proposal requires parentheses around arrow functions
and bans await from the pipeline. The `fsharpPipeline` flag will
be responsible for enabling the await-in-pipeline behaviors.

### CODE REVIEW

**Comment by mAAdhaTTah** [comment]
@js-choi For your review.

File: packages/babylon/src/parser/expression.js [hunk 1]
```diff
@@ -1960,23 +1960,31 @@ export default class ExpressionParser extends LValParser {
...
+      }
+
+      if (
+        this.match(tt.name) &&
+        this.state.value === "await" &&
+        this.state.inAsync
+      ) {
+        throw this.raise(
+          this.state.start,
+          `Unexpected "await" after pipeline body; await is banned in minimal proposal`,
```
Comment (js-choi) [inline packages/babylon/src/parser/expression.js]: I think this error message should be tweaked to say that bare pipeline bodies `|> await` are banned in the minimal proposal, not `await` in general. The minimal proposal does allow `await` as long as it’s not `|> await`. For instance, `x |> (await promiseThatWillResolveToFunction)`. I…think. See https://github.com/tc39/proposal-pipeline-operator/pull/108#issuecomment-373991539.

**Comment by js-choi** [comment]
The changes look good to me, with [one exception on which I’ve commented](https://github.com/valtech-nyc/babel/pull/2/files#diff-fca1cacaed72b5d331b53e0a5d46503cR1985).

> The `fsharpPipeline` flag will be responsible for enabling the await-in-pipeline behaviors. 

Have you come to a decision about the F-Sharp Proposal’s precedence between `|>` and `=>` for now? That’s another potential difference from the Minimal Proposal.

File: packages/babylon/src/parser/expression.js (hunk 1 above)
Comment (mAAdhaTTah) [reply packages/babylon/src/parser/expression.js]: Oh, right, I should check that:

```js
x |> (await p)
```

parses correctly, as that's still allowed.

**Comment by mAAdhaTTah** [comment]
> Have you come to a decision about the F-Sharp Proposal’s precedence between `|>` and `=>` for now? That’s another potential difference from the Minimal Proposal.

I haven't yet, but the current minimal requires it, so I'm punting that decision until I implement it. The parenthesized arrow functions is the only difference between F# and the other two, but F# relies on arrow functions to flesh out its usefulness, so I'm still trying to decide if the trade-off is worth it.

The other alternative would be to hitch F# to partial application and try and advance them both together. Parentheses may not be bad enough to worry about, and it does come with some unpleasant side effects (`const a = (x |> f)` & `const f = x => (x |> a.f)`).

File: packages/babylon/src/parser/expression.js (hunk 1 above)
Comment (mAAdhaTTah) [reply packages/babylon/src/parser/expression.js]: Changed the error message. it parses correctly as-is though.

**Comment by js-choi** [comment]
Understood. The pull request with 8bff3f329c8c57633b0328229c0d7acd0b515647 looks good to me.

**Pull request merged**